### ISSUE TITLE
Variable created in context manager is detected as object of context variable class

### ISSUE THREAD

**Issue by Draqun** [PyCQA/pylint#4917]
### Bug description
Hi.
I want report bug which occurs in one of my project. As it is described in topic pylint detect object of class Animal created in context manager as an object of context manager class (SQLUnitOfWork) `src\api\webapp\animal.py:26:15: E1101: Instance of 'SQLUnitOfWork' has no 'as_dict' member (no-member)`. As a result it report my object does not have a method. 

Source code (full code you can find [here](https://github.com/Draqun/animal-api))

```python
    with SQLUnitOfWork(config) as uow:
        animal = Animal(**data).add_animal_to_db(uow)
        if not animal:
            return {'status': 'Incorrect values', 'values': body}, 405
        return animal.as_dict(), 201  # pylint: disable=no-member
```

### Command used

```shell
pylint --rcfile=.cicd/config/pylint src/
```

### Pylint output

```shell
************* Module src.api.webapp.animal
src\api\webapp\animal.py:25:15: E1101: Instance of 'SQLUnitOfWork' has no 'as_dict' member (no-member)

------------------------------------------------------------------
Your code has been rated at 9.74/10 (previous run: 9.74/10, +0.00)
```

### Expected behavior

This is false positive so It should never happen.

### Pylint version

```shell
$ pylint --version
pylint 2.10.2
astroid 2.7.2
Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
```
